spec/hashie/mash_spec.rb

```diff
@@ -821,9 +821,16 @@ class SubMash < Hashie::Mash
         expect(mash.company_a.accounts.admin.password).to eq('secret')
       end
       it 'can override the value of aliases' do
-        expect do
-          Hashie::Mash.load('spec/fixtures/yaml_with_aliases.yml', aliases: false)
-        end.to raise_error Psych::BadAlias, /base_accounts/
+        require 'psych'
+        if Psych::VERSION >= '5'
```

Comment:
**discussion:** This makes me think we should add the forking behavior into Mash so that the errors are the same. It appears that we should be enabling alias parsing in Psych 5?

I know that this is disabled by default in newer versions due to security concerns. Since this feature should not be used to parse external YAML, I think this makes sense ... though we should also note it in the README.

In addition, it would be nice to rescue errors and wrap them in an internal error.

What do you think?

cc @dblock, @BobbyMcWho

Reply:
I think having a Hashie owned error makes sense, though it's probably technically a breaking change.

Reply:
I agree this is ugly, but I just wanted to make it work.  :)

This spec in particular is a little confusing - it fails for both Psych 4 and Psych 5 (as well as earlier versions of Psych) because an explicit `aliases: false` is being passed in for a file that uses aliases.  So I don't think this is an issue of needed to change the default passed options.

But in a failure case the underlying error will be exposed, and that does change between versions of Psych. Given that Mash allows you to use alternate parsers, not just YAML ones, it certainly makes sense to capture load/parse errors and convert them to a Hashie-specific parse exception.  I think it would make sense to be handled in the `perform` method of the parser.

Reply:
I think we can agree that this is a topic for a future PR?

Reply:
Btw, I don't think this is a reliable way to compare Psych version, see https://code.dblock.org/2020/07/16/comparing-version-numbers-in-ruby.html for the fun gotcha's :)